sequencer/src/api.rs

```diff
@@ -115,10 +122,7 @@ mod test {
             handle.hotshot.start_consensus().await;
         }
 
-        let mut options = Options::from(options::Http { port }).submit(Default::default());
-        if let Some(query) = query_opt {
-            options = options.query_fs(query);
-        }
+        let options = opt(Options::from(options::Http { port }).submit(Default::default()));
```

Comment:
This seems a bit tricky to reason about together with the function/closure argument. Maybe add a comment about what the closure should and should not (i guess change the port (?)) do?